charts/docker-hub-rate-limit-exporter-chart/templates/deployment.yaml

```diff
@@ -1,55 +1,59 @@
+{{- if .Values.deployment.enabled }}
 apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: {{ include "docker-hub-rate-limit-exporter.fullname" . }}
   labels:
     {{- include "docker-hub-rate-limit-exporter.labels" . | nindent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount }}
+  replicas: {{ .Values.deployment.replicaCount }}
   selector:
     matchLabels:
       {{- include "docker-hub-rate-limit-exporter.selectorLabels" . | nindent 6 }}
-      {{- if .Values.podLabels }}
-      {{- toYaml .Values.podLabels | nindent 6 }}
-      {{- end }}
   template:
     metadata:
+      labels:
+        {{- include "docker-hub-rate-limit-exporter.selectorLabels" . | nindent 8 }}
       annotations:
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
         checksum/secret: {{ include (print $.Template.BasePath "/secret.yaml") . | sha256sum }}
-        {{- with .Values.podAnnotations }}
-        {{- toYaml . | nindent 8 }}
+        {{- if .Values.podAnnotations }}
+        {{- toYaml .Values.podAnnotations | nindent 8 }}
         {{- end }}
-      labels:
-        {{- include "docker-hub-rate-limit-exporter.selectorLabels" . | nindent 8 }}
     spec:
-      {{- with .Values.imagePullSecrets }}
+      {{- if .Values.imagePullSecrets }}
       imagePullSecrets:
-        {{- toYaml . | nindent 8 }}
+{{ toYaml .Values.imagePullSecrets | indent 8 }}
       {{- end }}
       serviceAccountName: {{ include "docker-hub-rate-limit-exporter.serviceAccountName" . }}
+      {{- if .Values.securityContext }}
       securityContext:
-        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+{{ toYaml .Values.securityContext | indent 8 }}
+      {{- end }}
+      {{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName }}
+      {{- end }}
       containers:
         - name: {{ .Chart.Name }}
-          securityContext:
-            {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- with .Values.containerSecurityContext }}
+          securityContext: {{ toYaml . | nindent 12 }}
+          {{- end }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           ports:
-            - name: http
-              containerPort: {{ default 80 .Values.config.exporterPort }}
+            - name: {{ .Values.service.portName }}
+              containerPort: {{ .Values.service.port }}
               protocol: TCP
           livenessProbe:
             httpGet:
               path: /
-              port: http
-            timeoutSeconds: 5
+              port: {{ .Values.service.port }}
+            timeoutSeconds: {{ .Values.service.timeoutSeconds }}
```

Comment:
This is minor, but I find it confusing to have the `timeoutSeconds` as part of `.Values.service` since the service-object in kubernetes does not have a timeout property, I think it would be better to call it `.Values.probeTimeoutSeconds` or `.Values.livenessProbeProbeTimeoutSeconds` and `.Values.readinessProbeProbeTimeoutSeconds`.

What do you think about that?

Reply:
yes i prefer your method